firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -140,12 +145,12 @@ public static FirebaseInstallations getInstance(@NonNull FirebaseApp app) {
   @Override
   public synchronized Task<InstallationTokenResult> getAuthToken(
```

Comment:
we don't wanna synchronize the getAuthToken method, just the `addGetAuthTokenListener` method. This is also synchronized on `this` which doesn't help us; we need to synchronize on `lock`